saas/api/roles.py

```diff
@@ -670,8 +672,35 @@ class RoleByDescrQuerysetMixin(RoleDescriptionMixin, RoleQuerysetMixin):
 
     def get_queryset(self):
         return super(RoleByDescrQuerysetMixin, self).get_queryset().filter(
-            Q(role_description=self.role_description)
-            | Q(request_key__isnull=False))
+            role_description=self.role_description)
+
+
+class RoleListPagination(PageNumberPagination):
+
+    def paginate_queryset(self, queryset, request, view=None):
+        self.invited_count = queryset.filter(
+            grant_key__isnull=False).count()
+        self.requested_count = queryset.filter(
+            request_key__isnull=False).count()
+        qry = {}
```

Comment:
Code starting at this line to the end of the function is at the incorrect place. It should be in `get_queryset`.
The only intent of `paginate_queryset` is to return one page of results from `get_queryset`.
`paginate_queryset` is also used to decorate the returned `data` json with `count` which is used together with the number of results per page to show a page count.
Here you cannot use `self.invited_count = queryset.filter(grant_key__isnull=False).count()` because `invited` is not a direct subset of the queryset that is paginated. For `self.invited_count` and `self.requested_count`, you have to use different querysets than the one being paginated.